src/pages/property/[uid].js

```diff
@@ -52,125 +59,174 @@ function groupArr(data, n) {
+
+	useEffect(() => {
+		if (user) {
+			getPropertyListFromUserID(user.uid, property.uid)
+				.then((res) => {
+					setLike(res.includes(property.uid));
+				})
+				.catch((err) => {
+					console.log('err', err);
+				});
+		}
+	}, [user, property.uid]);
+
+	const handleLike = (e) => {
+		setLike(!like);
+		if (like) {
+			deleteUserPropertyMapping(user.uid, property.uid);
+		} else {
+			addUserPropertyPreference(user.uid, property.uid);
+		}
 	};
-	const x = [1, 2, 3, 4, 5, 6, 7, 8];
 
-	let groupsOf3 = groupArr(x, 4);
+	const x = [1, 2, 3];
 
 	return (
-		<Card sx={{ m: '80px', width: 'auto', backgroundColor: '#F8F8F2' }} onClick={() => callme()}>
-			<Grid
-				container
-				direction="row"
-				justifyContent="flex-start"
-				alignItems="center"
-				spacing={4}
-				sx={{ paddingTop: '40px' }}>
-				<Grid item xs={9}>
-					<Typography align="right" variant="h4" component="div">
-						{property.name}
-					</Typography>
-				</Grid>
-				<Grid align="center" item xs={3}>
-					<IconButton aria-label="add to favorites">
-						<FavoriteIcon fontSize="large" />
+		<Container maxWidth="false" sx={{ m: '50px', width: 'auto' }}>
+			<div style={{ backgroundColor: '#F8F8F2' }}>
+				<Stack direction="row" justifyContent="space-between" alignItems="center" spacing={2} py={3}>
+					<p></p>
+					<Typography variant="h4">{property.name}</Typography>
+					<IconButton aria-label="add to favorites" onClick={handleLike}>
+						{like ? <FavoriteIcon fontSize="large" sx={{ color: 'red' }} /> : <FavoriteIcon fontSize="large" />}
 					</IconButton>
-				</Grid>
-			</Grid>
+				</Stack>
+
+				<Divider />
+
+				<Grid container direction="row" justifyContent="space-around" alignItems="center" spacing={4}>
+					<Grid item xs={12} sm={6} md={5} xl={4}>
+						<Carousel navButtonsAlwaysVisible sx={{ maxWidth: '600px', margin: '50px' }}>
+							{x &&
+								x.map((obj, i) => {
+									return (
+										<CardMedia
+											sx={{ height: '100%', width: '100%' }}
+											component="img"
+											image={property.imageUrls[0]}
+											alt="property1.jpg"
+											key={i}
+										/>
+									);
+								})}
+						</Carousel>
+					</Grid>
 
-			<Grid container direction="row" justifyContent="space-around" alignItems="center" spacing={4}>
-				<Grid>
-					<Carousel navButtonsAlwaysVisible sx={{ width: '400px', margin: '50px' }}>
-						{x &&
-							x.map((obj, i) => {
-								return (
-									<CardMedia
-										sx={{ height: '100%', width: '100%' }}
-										component="img"
-										image={property.imageUrls[0]}
-										alt="property1.jpg"
-										key={i}
+					<Grid item xs={12} sm={6} md={7} xl={8}>
+						<Typography variant="h5" mt={3}>
+							Information
+						</Typography>
+						<Grid container direction="row" justifyContent="flex-start" alignItems="flex-start" spacing={1}>
+							<Grid item xs={12} md={4} lg={4}>
+								<PrefIcon
+									icon={<AttachMoneyIcon sx={{ fontSize: 40 }} />}
```

Comment:
not a big fan of writing `fontSize: 40` everywhere, I think the better approach would have been writing your own wrapper class around PrefIcon or something else to have fontSize 40 so we can change all the icons at once. if we decide we want 45 it's gonna take a lot more work than just updating one class. not a big enough deal to block the merge, but if you're copy pasting a ton it's a sign a class can be made

Reply:
Good observation and suggestion. Fixed.